### DISCUSSION TITLE
post_233451

### DISCUSSION THREAD

**Discussion by cincuranet** [cincuranet/blog-comments#1226] [Comments]
imported 

https://www.tabsoverspaces.com/id/233451

**Comment by cincuranet** [comment]
Imported comment written by _Bob Archer_ on _2014-04-24T19:54:27_:

<p>Thanks, that pretty much what I figured was needed. I'm not sure why this isn't built into the DbSet.</p>

**Comment by skalahonza** [comment]
There is one hidden gotcha tho.
Let's say that you will use the built in Find and then this method

````cs
var document = await context.Documents.FindAsync(123);
// this puts Document with id 123 into your local context
```

````cs
var document = await context.Documents.FindPredicate(x => true);
// this only returns document 123 right?
```

**Comment by cincuranet** [comment]
Regarding the code, yes.

**Comment by skalahonza** [comment]
That's the gotcha I mean. Because if you fetched some item matching the predicate before calling this extension method, you will not get any other items matching the predicate even if they are present in the database.

**Comment by cincuranet** [comment]
Correct. The local state takes priority.
